definitions/checks/katello/check_clean_backend_objects.rb

```diff
@@ -0,0 +1,33 @@
+module Checks::Katello
+  class CheckCleanBackendObjects < ForemanMaintain::Check
+    metadata do
+      description 'Check whether subscription consumers of backend service is in sync'
+      tags :pre_upgrade
+
+      confine do
+        feature(:katello)
+      end
+    end
+
+    def clean_backend_objects_dry_run
+      found_orphans = nil
+      list_cmd = "export RUBYOPT='-W0'; foreman-rake katello:clean_backend_objects"
```

Comment:
This command as shown here executes in a dry run mode? How does a user execute it in a non-dry run mode?

Reply:
These are automatically cleaned up as part of the upgrade? What is the goal of the check in this case? What are the actions a user should take?

Reply:
The problem was that customers wondered why many content hosts were suddenly deleted during upgrade run because candlepin subscriptions of hosts were no longer available. That's why we think that a warning in the pre-upgrade is good

Reply:
What is the expectation of customers hitting this and having their upgrade halted? To run the script by hand with commit? To skip this check after the first time they see it?

Reply:
Run the "foreman-maintain upgrade check" first - before they acutally run the upgrade. I suggest to extend this "pre-upgrade check" further in the future to make other things visible which hurts you during the upgrade step. 

If you run the "upgrade check" and make it visible, that hundreds of hosts will be unregistered/deleted during upgrade run, you have the chance to react accordingly - like to read some knowledge base articles to examine the root cause.

Reply:
Can we provide links or more action forward language for the user?

My concern is the user runs `upgrade check` or `upgrade` sees this step fail and then is left without any clear understanding of what they should do next. Should they ignore it? Should they read about it? Should they perform some action independently of the tool?

Reply:
I did miss some of the potential output of the rake-task.
I am not fully sure, what can be done about all the findings. I assume re-registering to the activation-key on the found hosts can fix this.
I am not aware of additional information, I can link to.
@ehelms do you know of any Discourse-threads or documentation-chapters that provide additional information?